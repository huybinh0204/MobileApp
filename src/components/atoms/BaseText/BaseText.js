import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import Animated from 'react-native-reanimated';

const BaseText = ({ children, variant, ...props }) => {
  const TextComponent = variant === 'animated' ? Animated.Text : Text;
  return <TextComponent {...props}>{children}</TextComponent>;
};

BaseText.propTypes = {
  allowFontScaling: PropTypes.bool,
  variant: PropTypes.any,
};

BaseText.defaultProps = {
  allowFontScaling: false,
  variant: 'normal',
};

export default BaseText;
