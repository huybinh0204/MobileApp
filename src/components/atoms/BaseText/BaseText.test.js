import { render } from '@testing-library/react-native';
import React from 'react';
import BaseText from './BaseText';

describe('Text', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<BaseText>test</BaseText>);

    expect(toJSON()).toMatchSnapshot();
  });
});
