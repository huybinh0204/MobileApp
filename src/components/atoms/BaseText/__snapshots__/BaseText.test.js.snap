// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match the snapshot 1`] = `
<Text
  allowFontScaling={false}
>
  test
</Text>
`;
